providers/shared/flows/components/flow.ftl

```diff
@@ -174,169 +181,186 @@
-                                context=link
-                            /]
-                            [#return {} ]
-                        [#else]
-                            [#local linkType = subComponent.Type ]
-                        [/#if]
-                        [#break]
-                    [/#if]
-                [/#list]
-                [#if subComponentId?has_content ]
-                    [#break]
-                [/#if]
-            [/#list]
-        [/#if]
+        [#local subOccurrences = getOccurrenceChildren(targetOccurrence) ]
 
         [#-- Legacy support for links to lambda without explicit function --]
-        [#-- TODO(mfl): Review legacy support with view to removal --]
-        [#if hasOccurrenceChildren(targetOccurrence) &&
-                subComponentId == "" &&
-                (componentType == LAMBDA_COMPONENT_TYPE) ]
-            [#local subComponentId = (getOccurrenceChildren(targetOccurrence)[0].Core.SubComponent.Id)!"" ]
-            [#local linkType = LAMBDA_FUNCTION_COMPONENT_TYPE ]
-        [/#if]
+        [#-- Default to the first suboccurrence returned                  --]
+        [#-- TODO(mfl): Review legacy support with view to removal        --]
+        [#if subOccurrences?has_content &&
+                effectiveLink.SubComponent == "" &&
+                (getOccurrenceType(targetOccurrence) == LAMBDA_COMPONENT_TYPE) ]
+            [#local effectiveLink +=
+                {
+                    "Type" : LAMBDA_FUNCTION_COMPONENT_TYPE,
+                    "SubComponent" : (subOccurrences[0].Core.SubComponent.Id)!""
+                }
+            ]
+            [#local matchedOccurrence = subOccurrences[0] ]
+        [#else]
+
+            [#local matchedOccurrence = targetOccurrence ]
 
-        [#-- Find suboccurrences with a matching id --]
-        [#if subComponentId?has_content]
-            [#local potentialOccurrences = [] ]
-            [#list getOccurrenceChildren(targetOccurrence) as potentialOccurrence]
-                [#if
-                    (subComponentId == (getOccurrenceSubComponent(potentialOccurrence).Id)!"") &&
-                    (getOccurrenceInstance(potentialOccurrence).Id == instanceToMatch) &&
-                    (getOccurrenceVersion(potentialOccurrence).Id == versionToMatch) &&
-                    (
-                        (!linkType?has_content) ||
-                        (linkType == getOccurrenceType(potentialOccurrence))
-                    ) ]
-
-                    [#local potentialOccurrences += [potentialOccurrence] ]
+            [#if effectiveLink.SubComponent?has_content]
+
+                [#-- Match is to a SubComponent --]
+
+                [#if subOccurrences?size > 1 ]
+                    [@fatal
+                        message='Multiple matching subcomponents with id "${effectiveLink.SubComponent}". Use a Type attribute to differentiate them.'
+                        context=subOccurrences
```

Comment:
Same here, maybe some specific details rather than the occurrence details

Reply:
see above